syft/frameworks/torch/tensors/interpreters/native.py

```diff
@@ -992,19 +992,59 @@ def torch_type(self):
         else:
             return self.child.torch_type()
 
-    def encrypt(self, public_key):
-        """This method will encrypt each value in the tensor using Paillier
-        homomorphic encryption.
+    def encrypt(self, protocol="mpc", **kwargs):
+        """
+        This method will encrypt each value in the tensor using Multi Party
+        Computation (default) or Paillier Homomorphic Encryption
 
         Args:
-            *public_key a public key created using
-                syft.frameworks.torch.he.paillier.keygen()
+            protocol (str): Currently supports 'mpc' for Multi Party
+                Computation and 'paillier' for Paillier Homomorphic Encryption
+            **kwargs:
+                With Respect to MPC accepts:
+                    workers (list): Parties involved in the sharing of the Tensor
+                    crypto_provider (syft.VirtualWorker): Worker responsible for the
+                        generation of the random numbers for encryption
+                    Keyword Args: To be parsed as kwargs for the .fix_prec() method
+
+                With Respect to Paillier accepts:
+                    public_key (phe.paillier.PaillierPublicKey): Can be obtained using
+                        ```public_key, private_key = sy.frameworks.torch.he.paillier.keygen()```
+        Returns:
+            An encrypted version of the Tensor following the protocol specified
+
+        Raises:
+            NotImplementedError: If protocols other than the ones mentioned above are queried
+
         """
+        if protocol.lower() == "mpc":
+            workers = kwargs.pop("workers")
+            crypto_provider = kwargs.pop("crypto_provider")
+            kwargs_fix_prec = kwargs  # Rest of kwargs for fix_prec method
+
+            if kwargs_fix_prec:
+                x_shared = self.fix_prec(**kwargs_fix_prec).share(
+                    *workers, crypto_provider=crypto_provider
+                )
+            else:  # If no kwargs for .fix_prec()
+                x_shared = self.fix_prec().share(*workers, crypto_provider=crypto_provider)
```

Comment:
I don't think this case is different from the one above, is it?
Doesn't `self.fix_prec(**kwargs_fix_prec)` work with an empty dict?

Reply:
Yes. You are correct. 
That style was necessary for an older version of code where the kwargs were "None" rather than an empty dict.